### DISCUSSION TITLE
Is it possible to "fire and forget"?

### DISCUSSION THREAD

**Discussion by Kurren123** [Azure/azure-functions-durable-extension#2226] [Q&A] [answered]
I want to wait for an external event and send a reminder if the event hasn't occurred in 5 days. Sending the reminder will be a "fire and forget" task. I am new to how tasks work. Will the following orchestrator function be okay? I didn't use `Task.WhenAll` as I thought this was simpler:

```csharp
[FunctionName("Function1")]
public static async Task RunOrchestrator(
    [OrchestrationTrigger] IDurableOrchestrationContext context)
{
    using (var cts = new CancellationTokenSource())
    {
        SendReminder(context, cts.Token); // dont await so this happens in the background
        await context.WaitForExternalEvent("MyEvent");
        cts.Cancel(); // cancel the reminder as the event has now happened
        
        // react to MyEvent ...

    }
}

public static async Task SendReminder(IDurableOrchestrationContext context, CancellationToken cancellationToken)
{
    await context.CreateTimer(context.CurrentUtcDateTime.AddDays(5), cancellationToken);
    await context.CallHttpAsync(...) // http call to ms graph to send a reminder email
}
```

I have read the [code constraints article](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-code-constraints?tabs=csharp) so I know not to use `Task.Run` or `configureAwait(false)`, but is it okay just calling `SendReminder` without `await`?

**Comment by cgillum** [comment, marked as answer]
I think this should be okay. I need to be reminded about what the behavior is when cancelling a timer (it might result in a `TaskCancelledException` at the point of `await context.CreateTimer(...)`, but otherwise I think what you have is fine.

I don't think this applies to your example here, but one behevior to be aware of is that all outstanding tasks must be completed or canceled before an orchestration can transition into a `Completed` state, so if you execute a task, don't await it, and let the function complete, you might be surprised to see that your orchestration remains in the `Running` state until the non-awaited tasks are completed.
